src/main/java/io/vertx/lang/jruby/JRubyVerticleFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.vertx.lang.jruby;
+
+import io.vertx.core.Verticle;
+import io.vertx.core.Vertx;
+import io.vertx.core.spi.VerticleFactory;
+import org.jruby.CompatVersion;
+import org.jruby.RubyInstanceConfig;
+import org.jruby.embed.LocalContextScope;
+import org.jruby.embed.ScriptingContainer;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ */
+public class JRubyVerticleFactory implements VerticleFactory {
+
+
+  private ScriptingContainer container;
+  private Vertx vertx;
+
+  @Override
+  public String prefix() {
+    return "rb";
+  }
+
+  @Override
+  public void init(Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  @Override
+  public Verticle createVerticle(String verticleName, ClassLoader classLoader) throws Exception {
+    return new JRubyVerticle(this, classLoader, verticleName);
+  }
+
+  ScriptingContainer createContainer(String gemPath) {
+    ScriptingContainer container = new ScriptingContainer(LocalContextScope.SINGLETHREAD);
```

Comment:
You'll also need to set compat mode 1.9, and create it with the right context classloader set, and set the error writer.

Copy how it's done in the Vert.x 2 mod-lang-jruby if in doubt